python/hsfs/feature_group.py

```diff
@@ -216,6 +216,68 @@ def get_tags(self):
         """
         return self._feature_group_engine.get_tags(self)
 
+    def get_parent_feature_groups(self):
+        """Get the parent feature groups that were used to generate this feature
+        group. This is based on explicit provenance.
+        """
+        links_dto = self._feature_group_engine.get_explicit_provenance_links(self)
+        parent_links = []
+        for link in links_dto.upstream:
+            parent_links.append(
+                {
+                    "project": link["project"],
+                    "name": link["name"],
+                    "version": link["version"],
+                    "type": link["type"],
+                    "accessible": link["accessible"],
+                    "deleted": link["deleted"],
+                }
+            )
+        return parent_links
+
+    def get_derived_feature_groups(self):
+        """Get the derived feature groups that used this feature group.
+        This is based on explicit provenance.
+        """
+        links_dto = self._feature_group_engine.get_derived_artifacts(self)
+        derived_feature_groups = []
+        for link in links_dto.upstream:
+            if (
+                link["type"] == "CACHED_FEATURE_GROUP"
+                or link["type"] == "FEATURE_GROUP"
```

Comment:
What's the difference between these 2? Should it be `STREAM_FEATURE_GROUP`? What about the external feature groups?

Reply:
typo, it was indeed STREAM_FEATURE_GROUP.